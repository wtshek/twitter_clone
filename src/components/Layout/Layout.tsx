import { FC, ReactNode, useEffect, useState } from "react";
import cx from "classnames";
import { BottomNavBar } from "./BottomNavBar";
import { Header } from "./Header";
import SideMenu from "./SideMenu/SideMenu";

type LayoutProps = {
  children: ReactNode;
};

export const Layout: FC<LayoutProps> = ({ children }) => {
  const [isSideMenuOpen, setIsSideMenuOpen] = useState(false);

  const handleOpenSideMenu = () => {
    setIsSideMenuOpen(true);
  };

  const handleCloseSideMenu = () => {
    setIsSideMenuOpen(false);
  };

  return (
    <div>
      <Header handleOpenSideMenu={handleOpenSideMenu} />
      {children}
      <BottomNavBar />
      <SideMenu
        handleCloseSideMenu={handleCloseSideMenu}
        containerClassName={cx("duration-100", {
          "-translate-x-full": !isSideMenuOpen,
        })}
        overlayClassName={cx("duration-100", {
          "opacity-0 -translate-x-full": !isSideMenuOpen,
        })}
      />
    </div>
  );
};

export default Layout;
