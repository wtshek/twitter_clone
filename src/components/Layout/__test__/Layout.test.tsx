import { render } from "@testing-library/react";
import Layout from "../Layout";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  Link: ({ children }: { children: JSX.Element }) => <div>{children}</div>,
  useLocation: () => "/",
}));

describe("Layout component", () => {
  it("should match the snapshot", () => {
    const { container } = render(
      <Layout>
        <div>Layout Test</div>
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });
});
