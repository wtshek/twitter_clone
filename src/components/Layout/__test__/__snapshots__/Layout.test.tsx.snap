// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="fixed w-full pt-5 px-4 bg-transparent-background backdrop-blur-md duration-200 border-b-[1px] border-dark-border translate-y-0"
    >
      <div
        class="relative flex justify-center items-center mb-4"
      >
        <button
          class="absolute left-4"
        >
          <div
            class="rounded-full object-cover"
          >
            <img
              alt="avatar"
              height="32"
              src="/default_avatar.png"
              width="32"
            />
          </div>
        </button>
        <svg
          class="w-23px"
          viewBox="0 0 248 204"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M221.95 51.29c.15 2.17.15 4.34.15 6.53 0 66.73-50.8 143.69-143.69 143.69v-.04c-27.44.04-54.31-7.82-77.41-22.64 3.99.48 8 .72 12.02.73 22.74.02 44.83-7.61 62.72-21.66-21.61-.41-40.56-14.5-47.18-35.07 7.57 1.46 15.37 1.16 22.8-.87-23.56-4.76-40.51-25.46-40.51-49.5v-.64c7.02 3.91 14.88 6.08 22.92 6.32C11.58 63.31 4.74 33.79 18.14 10.71c25.64 31.55 63.47 50.73 104.08 52.76-4.07-17.54 1.49-35.92 14.61-48.25 20.34-19.12 52.33-18.14 71.45 2.19 11.31-2.23 22.15-6.38 32.07-12.26-3.77 11.69-11.66 21.62-22.2 27.93 10.01-1.18 19.79-3.86 29-7.95-6.78 10.16-15.32 19.01-25.2 26.16z"
            fill="#1d9bf0"
          />
        </svg>
      </div>
      <div
        class="flex"
      >
        <button
          class="relative text-base text-white font-bold w-1/2 h-[53px]"
        >
          <div
            class="inline-flex flex-col justify-around h-full"
          >
            <span>
              For You
            </span>
            <span
              class="h-1 bg-main w-full rounded-full"
            />
          </div>
        </button>
        <button
          class="relative text-base text-white font-bold w-1/2 h-[53px] text-dark-grey"
        >
          <div
            class="inline-flex flex-col justify-around h-full"
          >
            <span>
              Following
            </span>
            <span
              class="h-1 bg-main w-full rounded-full opacity-0"
            />
          </div>
        </button>
      </div>
    </div>
    <div>
      Layout Test
    </div>
    <div
      class="fixed bottom-0 inset-x-0 flex items-center justify-around border-t-[1px] border-dark-border bg-dark-background"
    >
      <div>
        <svg
          aria-hidden="true"
          class="fill-white w-26px"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M12 1.696L.622 8.807l1.06 1.696L3 9.679V19.5C3 20.881 4.119 22 5.5 22h13c1.381 0 2.5-1.119 2.5-2.5V9.679l1.318.824 1.06-1.696L12 1.696zM12 16.5c-1.933 0-3.5-1.567-3.5-3.5s1.567-3.5 3.5-3.5 3.5 1.567 3.5 3.5-1.567 3.5-3.5 3.5z"
            />
          </g>
        </svg>
      </div>
      <div>
        <svg
          aria-hidden="true"
          class="fill-white w-26px"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M10.25 3.75c-3.59 0-6.5 2.91-6.5 6.5s2.91 6.5 6.5 6.5c1.795 0 3.419-.726 4.596-1.904 1.178-1.177 1.904-2.801 1.904-4.596 0-3.59-2.91-6.5-6.5-6.5zm-8.5 6.5c0-4.694 3.806-8.5 8.5-8.5s8.5 3.806 8.5 8.5c0 1.986-.682 3.815-1.824 5.262l4.781 4.781-1.414 1.414-4.781-4.781c-1.447 1.142-3.276 1.824-5.262 1.824-4.694 0-8.5-3.806-8.5-8.5z"
            />
          </g>
        </svg>
      </div>
      <div>
        <svg
          aria-hidden="true"
          class="fill-white w-26px"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M19.993 9.042C19.48 5.017 16.054 2 11.996 2s-7.49 3.021-7.999 7.051L2.866 18H7.1c.463 2.282 2.481 4 4.9 4s4.437-1.718 4.9-4h4.236l-1.143-8.958zM12 20c-1.306 0-2.417-.835-2.829-2h5.658c-.412 1.165-1.523 2-2.829 2zm-6.866-4l.847-6.698C6.364 6.272 8.941 4 11.996 4s5.627 2.268 6.013 5.295L18.864 16H5.134z"
            />
          </g>
        </svg>
      </div>
      <div>
        <svg
          aria-hidden="true"
          class="fill-white w-26px"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M1.998 5.5c0-1.381 1.119-2.5 2.5-2.5h15c1.381 0 2.5 1.119 2.5 2.5v13c0 1.381-1.119 2.5-2.5 2.5h-15c-1.381 0-2.5-1.119-2.5-2.5v-13zm2.5-.5c-.276 0-.5.224-.5.5v2.764l8 3.638 8-3.636V5.5c0-.276-.224-.5-.5-.5h-15zm15.5 5.463l-8 3.636-8-3.638V18.5c0 .276.224.5.5.5h15c.276 0 .5-.224.5-.5v-8.037z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="fixed top-0 left-0 bg-dark-overlay w-full opacity-1 duration-100 opacity-0 -translate-x-full"
    >
      <div
        class="w-3/4 h-screen overflow-y-auto bg-dark-background duration-100 -translate-x-full"
      >
        <div
          class="flex justify-between px-4 pb-8 mt-3"
        >
          <span
            class="text-white font-bold"
          >
            Account Info
          </span>
          <button>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-inverse "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex flex-col"
        >
          <div
            class="px-4"
          >
            <img
              alt="Avatar"
              class="w-10 h-10 rounded-full"
              src="https://via.placeholder.com/100x100"
            />
            <div>
              <div
                class="text-white text-lg font-medium"
              >
                John Doe
              </div>
              <div
                class="text-sm text-dark-grey"
              >
                @JDoe
              </div>
              <div
                class="flex mt-3"
              >
                <div
                  class="text-dark-grey mr-5 text-sm"
                >
                  <div
                    class="text-white font-medium"
                  >
                    0
                  </div>
                   Followings
                </div>
                <div
                  class="text-dark-grey text-sm"
                >
                  <div
                    class="text-white font-medium"
                  >
                    0
                  </div>
                   Followers
                </div>
              </div>
            </div>
          </div>
          <div
            class="mt-6"
          >
            <button
              class="flex items-center p-4"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-user fa-inverse mr-6"
                data-icon="user"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="text-white text-xl font-bold"
              >
                Profile
              </span>
            </button>
            <button
              class="flex items-center p-4"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-bookmark fa-inverse mr-6"
                data-icon="bookmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 48V487.7C0 501.1 10.9 512 24.3 512c5 0 9.9-1.5 14-4.4L192 400 345.7 507.6c4.1 2.9 9 4.4 14 4.4c13.4 0 24.3-10.9 24.3-24.3V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="text-white text-xl font-bold"
              >
                Topics
              </span>
            </button>
            <button
              class="flex items-center p-4"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-list fa-inverse mr-6"
                data-icon="list"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M40 48C26.7 48 16 58.7 16 72v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V72c0-13.3-10.7-24-24-24H40zM192 64c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zm0 160c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zm0 160c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zM16 232v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V232c0-13.3-10.7-24-24-24H40c-13.3 0-24 10.7-24 24zM40 368c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V392c0-13.3-10.7-24-24-24H40z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="text-white text-xl font-bold"
              >
                Bookmarks
              </span>
            </button>
            <button
              class="flex items-center p-4"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-table-cells fa-inverse mr-6"
                data-icon="table-cells"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm88 64v64H64V96h88zm56 0h88v64H208V96zm240 0v64H360V96h88zM64 224h88v64H64V224zm232 0v64H208V224h88zm64 0h88v64H360V224zM152 352v64H64V352h88zm56 0h88v64H208V352zm240 0v64H360V352h88z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="text-white text-xl font-bold"
              >
                Lists
              </span>
            </button>
            <button
              class="flex items-center p-4"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-table-cells fa-inverse mr-6"
                data-icon="table-cells"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm88 64v64H64V96h88zm56 0h88v64H208V96zm240 0v64H360V96h88zM64 224h88v64H64V224zm232 0v64H208V224h88zm64 0h88v64H360V224zM152 352v64H64V352h88zm56 0h88v64H208V352zm240 0v64H360V352h88z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="text-white text-xl font-bold"
              >
                Twitter Circle
              </span>
            </button>
          </div>
          <hr
            class="border-dark-border w-[89%] ml-4 mb-1"
          />
          <div>
            <button
              class="flex items-start w-full p-4"
            >
              <span
                class="text-white text-lg font-bold text-[15px]"
              >
                Creator Studio
              </span>
            </button>
            <div
              class="hidden"
            >
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-table-cells fa-inverse mr-3"
                  data-icon="table-cells"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm88 64v64H64V96h88zm56 0h88v64H208V96zm240 0v64H360V96h88zM64 224h88v64H64V224zm232 0v64H208V224h88zm64 0h88v64H360V224zM152 352v64H64V352h88zm56 0h88v64H208V352zm240 0v64H360V352h88z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Analytics
                </span>
              </a>
            </div>
          </div>
          <div>
            <button
              class="flex items-start w-full p-4"
            >
              <span
                class="text-white text-lg font-bold text-[15px]"
              >
                Professional Tools
              </span>
            </button>
            <div
              class="hidden"
            >
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-wrench fa-inverse mr-3"
                  data-icon="wrench"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 320c88.4 0 160-71.6 160-160c0-15.3-2.2-30.1-6.2-44.2c-3.1-10.8-16.4-13.2-24.3-5.3l-76.8 76.8c-3 3-7.1 4.7-11.3 4.7H336c-8.8 0-16-7.2-16-16V118.6c0-4.2 1.7-8.3 4.7-11.3l76.8-76.8c7.9-7.9 5.4-21.2-5.3-24.3C382.1 2.2 367.3 0 352 0C263.6 0 192 71.6 192 160c0 19.1 3.4 37.5 9.5 54.5L19.9 396.1C7.2 408.8 0 426.1 0 444.1C0 481.6 30.4 512 67.9 512c18 0 35.3-7.2 48-19.9L297.5 310.5c17 6.2 35.4 9.5 54.5 9.5zM80 408a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Twitter for Professionals
                </span>
              </a>
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-wrench fa-inverse mr-3"
                  data-icon="wrench"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 320c88.4 0 160-71.6 160-160c0-15.3-2.2-30.1-6.2-44.2c-3.1-10.8-16.4-13.2-24.3-5.3l-76.8 76.8c-3 3-7.1 4.7-11.3 4.7H336c-8.8 0-16-7.2-16-16V118.6c0-4.2 1.7-8.3 4.7-11.3l76.8-76.8c7.9-7.9 5.4-21.2-5.3-24.3C382.1 2.2 367.3 0 352 0C263.6 0 192 71.6 192 160c0 19.1 3.4 37.5 9.5 54.5L19.9 396.1C7.2 408.8 0 426.1 0 444.1C0 481.6 30.4 512 67.9 512c18 0 35.3-7.2 48-19.9L297.5 310.5c17 6.2 35.4 9.5 54.5 9.5zM80 408a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Twitter Ads
                </span>
              </a>
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-wrench fa-inverse mr-3"
                  data-icon="wrench"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 320c88.4 0 160-71.6 160-160c0-15.3-2.2-30.1-6.2-44.2c-3.1-10.8-16.4-13.2-24.3-5.3l-76.8 76.8c-3 3-7.1 4.7-11.3 4.7H336c-8.8 0-16-7.2-16-16V118.6c0-4.2 1.7-8.3 4.7-11.3l76.8-76.8c7.9-7.9 5.4-21.2-5.3-24.3C382.1 2.2 367.3 0 352 0C263.6 0 192 71.6 192 160c0 19.1 3.4 37.5 9.5 54.5L19.9 396.1C7.2 408.8 0 426.1 0 444.1C0 481.6 30.4 512 67.9 512c18 0 35.3-7.2 48-19.9L297.5 310.5c17 6.2 35.4 9.5 54.5 9.5zM80 408a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Monetization
                </span>
              </a>
            </div>
          </div>
          <div>
            <button
              class="flex items-start w-full p-4"
            >
              <span
                class="text-white text-lg font-bold text-[15px]"
              >
                Settings and Support
              </span>
            </button>
            <div
              class="hidden"
            >
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-wrench fa-inverse mr-3"
                  data-icon="wrench"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 320c88.4 0 160-71.6 160-160c0-15.3-2.2-30.1-6.2-44.2c-3.1-10.8-16.4-13.2-24.3-5.3l-76.8 76.8c-3 3-7.1 4.7-11.3 4.7H336c-8.8 0-16-7.2-16-16V118.6c0-4.2 1.7-8.3 4.7-11.3l76.8-76.8c7.9-7.9 5.4-21.2-5.3-24.3C382.1 2.2 367.3 0 352 0C263.6 0 192 71.6 192 160c0 19.1 3.4 37.5 9.5 54.5L19.9 396.1C7.2 408.8 0 426.1 0 444.1C0 481.6 30.4 512 67.9 512c18 0 35.3-7.2 48-19.9L297.5 310.5c17 6.2 35.4 9.5 54.5 9.5zM80 408a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Settings and privacy
                </span>
              </a>
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle-question fa-inverse mr-3"
                  data-icon="circle-question"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24V250.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1H222.6c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Help Center
                </span>
              </a>
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-wrench fa-inverse mr-3"
                  data-icon="wrench"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 320c88.4 0 160-71.6 160-160c0-15.3-2.2-30.1-6.2-44.2c-3.1-10.8-16.4-13.2-24.3-5.3l-76.8 76.8c-3 3-7.1 4.7-11.3 4.7H336c-8.8 0-16-7.2-16-16V118.6c0-4.2 1.7-8.3 4.7-11.3l76.8-76.8c7.9-7.9 5.4-21.2-5.3-24.3C382.1 2.2 367.3 0 352 0C263.6 0 192 71.6 192 160c0 19.1 3.4 37.5 9.5 54.5L19.9 396.1C7.2 408.8 0 426.1 0 444.1C0 481.6 30.4 512 67.9 512c18 0 35.3-7.2 48-19.9L297.5 310.5c17 6.2 35.4 9.5 54.5 9.5zM80 408a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Data saver
                </span>
              </a>
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-wrench fa-inverse mr-3"
                  data-icon="wrench"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 320c88.4 0 160-71.6 160-160c0-15.3-2.2-30.1-6.2-44.2c-3.1-10.8-16.4-13.2-24.3-5.3l-76.8 76.8c-3 3-7.1 4.7-11.3 4.7H336c-8.8 0-16-7.2-16-16V118.6c0-4.2 1.7-8.3 4.7-11.3l76.8-76.8c7.9-7.9 5.4-21.2-5.3-24.3C382.1 2.2 367.3 0 352 0C263.6 0 192 71.6 192 160c0 19.1 3.4 37.5 9.5 54.5L19.9 396.1C7.2 408.8 0 426.1 0 444.1C0 481.6 30.4 512 67.9 512c18 0 35.3-7.2 48-19.9L297.5 310.5c17 6.2 35.4 9.5 54.5 9.5zM80 408a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Display
                </span>
              </a>
              <a
                class="flex items-center p-3"
                href="#"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-right-from-bracket fa-inverse mr-3"
                  data-icon="right-from-bracket"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="text-white text-[15px]"
                >
                  Log out
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
